### ISSUE TITLE
User can delete unpublished stories

### ISSUE THREAD

**Issue by bpawlak** [denverfoundation/storybase#374]
My current list of stories looks like:

Untitled Story 273b8841adb54ba19c998094f0f39afc
Untitled Story 34f1de723fb342b594dcc458610c25fc
Untitled Story e630903f0e1c4f29a9689ee92766eb3b
Untitled Story 053f4a85424f414b8307d79e25e52807
fdasf
My Test Story
Untitled Story 9a2e9da98a714b44b9c8a78432731041
Untitled Story ceac7cedfb5048e5a8de12d735409aef
Untitled Story 6d5bf1a575d046fb96570495c46cb610
Untitled Story db4a6f037958441a9e30095cc1c1831d

Obviously, it's from testing and general mucking about, but at some point I'd like to get rid of all of those false-starts and clear out my list of drafts.

**Comment by ghing** [comment]
@bpawlak Not right now, but this would be relatively easy to wire up.  To me the logical place for this to happen seems like the story list. Is that where you think it should live?

**Comment by bpawlak** [comment]
Yep.  

Could either be a single [delete] link after each unpublished story or a set of checkboxes with an action button that would allow multiple deletions

[x]  Storytobedeleted
[x]  Storytobedeleted
[  ]  Don'tdelete
[x]  Storytobedeleted

---

(Delete Selected)  <- button

If we go this route, however, let's talk about what other bulk actions a user might want to take against a selection of stories (if anything).  Then, we wouldn't just have a button on the bottom, but a function select with an "Apply" button

**Comment by patternleaf** [comment]
68bae65 includes a delete button (see #460), however deleted stories just wind up at the bottom of the list. @ghing, i figured the best place to filter them was at the query site (rather than say in the view), but wanted to check in with you on that first.

**Comment by ghing** [comment]
You have two choices for this.
1. Alter the view context

In `AcountStoriesView` define a `get context data` method that looks something like this:

```
def get_context_data(self, request):
        context = super(AccountStoriesView, self).get_context_data(**kwargs)
        context['stories'] = request.user.stories.exclude(status='deleted')
        return context
```

Then in the template, you'll do `{% for story in stories %}` instead of `{% for story in user.stories.all %}`

Check out the [docs for `TemplateView`](https://docs.djangoproject.com/en/dev/ref/class-based-views/base/#django.views.generic.base.TemplateView) for more on this.
1. Define a method on `storybase_story.managers.StoryManager` that excludes the deleted stories from the queryset (e.g. non_deleted_stories, or some name that's better than that crappy one).  You'll probably also need to add the [use_for_related_fields](https://docs.djangoproject.com/en/dev/topics/db/managers/#controlling-automatic-manager-types) attribute on `StoryManager`.  Then in the template you could do `{% for story in user.stories.non_deleted_stories %}`.

**Comment by patternleaf** [comment]
closed via 3a42627.
